http/http.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2014 Space Monkey, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import (
+	"net/http"
+)
+
+// Client is an interface that matches an http.Client
+type Client interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type responseWriterObserver struct {
+	w  http.ResponseWriter
+	sc *int
+}
+
+func (w *responseWriterObserver) WriteHeader(status_code int) {
+	w.sc = &status_code
```

Comment:
I would use a bool instead of a pointer to indicate whether the response has been written. A tiny bit more GC friendly.